### DISCUSSION TITLE
Transit Duration Parameterization?

### DISCUSSION THREAD

**Discussion by eas342** [exoplanet-dev/jaxoplanet#246] [Q&A] [answered]
I noticed that the orbits.transit parameterizes the lightcurve in terms of duration.
https://jax.exoplanet.codes/en/latest/autoapi/jaxoplanet/orbits/transit/
Traditionally, I have seen transiting planets parameterized in terms of semi-major axis divided by the stellar radius, a/R*. Is there a reason duration might be better to sample? I'm wondering how one might deal with a non-circular orbit though, which has a non-trivial transit duration.

**Comment by soichiro-hattori** [comment, marked as answer]
Hi @eas342,

I think you should be able to parameterize your orbit with a/R* if you use the [Keplerian orbits!](https://github.com/exoplanet-dev/jaxoplanet/blob/main/src/jaxoplanet/orbits/keplerian.py)!
The first part of the [Quickstart tutorial](https://jax.exoplanet.codes/en/latest/tutorials/quickstart/) might be what you're looking for?

**Comment by eas342** [comment]
Thank you @soichiro-hattori !

Ah, so for this a/R* parameterization and lightcurves, one should use `from jaxoplanet.experimental.starry.light_curves import light_curve` instead of `from jaxoplanet.light_curves import limb_dark_light_curve`? Does it do limb darkening and is it ready to be used, despite saying "experimental"?

**Reply by soichiro-hattori** [reply]
Sorry for a late response @eas342. 

> Ah, so for this a/R* parameterization and lightcurves, one should use from jaxoplanet.experimental.starry.light_curves import light_curve instead of from jaxoplanet.light_curves import limb_dark_light_curve?

You should be able to still use `from jaxoplanet.light_curves.limb_dark import light_curve`! The [Occultation light curve of limb-darkened star](https://jax.exoplanet.codes/en/latest/tutorials/quickstart/#occultation-light-curve-of-limb-darkened-star) section of the Quickstart tutorial shows an example! 

> Does it do limb darkening and is it ready to be used, despite saying "experimental"?

It does LD and this part is actually not in the "experimental" `starry` part. Also, @lgrcia can chime in more about this but the `starry` part should be pretty mature and ready to be moved out of "experimental"!

**Reply by eas342** [reply]
Super, thank you @soichiro-hattori ! Based on the information you sent, I made this short tutorial for myself, to match the Quick Start tutorial of the commonly-used `batman` package:
https://lkreidberg.github.io/batman/docs/html/quickstart.html


    from jaxoplanet.orbits.keplerian import System, Central
    import astropy.units as u
    import astropy.constants as const
    from jaxoplanet.units import unit_registry as ureg
    from jaxoplanet.light_curves.limb_dark import light_curve
    import numpy as np
    import jax.numpy as jnp
    from matplotlib import pyplot as plt

    period = 1.0 * u.day## days
    rp = 0.1 ## in Rp/R*
    a = 15. ## in a/R*
    inc = 87. * ureg.deg ## in degrees

    # Use Kepler's 3rd law to calculate a mass given the a/R* and period
    
    mass = (2. * np.pi / (period))**2 * (a * u.Rsun)**3 / const.G
    #unitless_prefac = prefac.to(u.Msun).value

    star = Central(
        radius=1.0 * ureg.R_sun,
        mass = mass.to(u.Msun),
    )

    system = System(star).add_body(
        period = period,
        radius=rp * ureg.R_sun,
        mass=0.0 * ureg.M_sun,
        inclination=inc,
    )

    #planet = system.bodies[0]

    # checking the parameters of the system
    #system

    u = [0.1, 0.3]
    time = jnp.linspace(-0.05, 0.05, 1000)

    flux = 1.0 + light_curve(system, u)(time)


    plt.plot(time,flux)
    plt.xlabel("Time from central transit")
    plt.ylabel("Relative flux")

![Screenshot 2025-01-24 at 4 30 33 PM](https://github.com/user-attachments/assets/9ee7805f-4d3f-4fde-857a-fd50329de8ef)

Looks like the lightcurves agree with Batman to 0.1 ppm


Do you think this might be useful enough I should put it in the docs and do a pull request?
